src/blueapi/client/rest.py

```diff
@@ -127,10 +134,22 @@ def _request_and_deserialize(
         get_exception: Callable[[requests.Response], Exception | None] = _exception,
     ) -> T:
         url = self._url(suffix)
+        headers: dict[str, str] = {
+            "content-type": "application/json; charset=UTF-8",
+        }
+        if self._session_manager and (token := self._session_manager.get_token()):
+            try:
+                self._session_manager.authenticator.decode_jwt(token["access_token"])
+                headers["Authorization"] = f"Bearer {token['access_token']}"
+            except jwt.ExpiredSignatureError:
+                if token := self._session_manager.refresh_auth_token():
+                    headers["Authorization"] = f"Bearer {token['access_token']}"
+            except Exception:
+                pass
```

Comment:
Unless there's an exception here that we explicitly want to ignore (in which case we should document it) we should raise whatever issue is here. In the case that config isn't enabled we'll not enter this branch so don't need to worry about that case being unhappy path.

```suggestion
```